aten/src/ATen/native/cuda/TensorFactories.cu

```diff
@@ -103,28 +104,41 @@ Tensor& randperm_out_cuda(Tensor& result, int64_t n, c10::optional<Generator> ge
   // Generate random values for the keys array
   AT_DISPATCH_ALL_TYPES(
     result.scalar_type(), "randperm_out_cuda", [&] {
+      TORCH_CHECK(n <= std::numeric_limits<int>::max(),
+        "randperm of tensors larger than INT_MAX is not supported yet in pytorch");
+
       auto keys = at::empty(result.sizes(), result.options()).random_(generator);
-      auto keys_data = thrust::device_ptr<scalar_t>(keys.data_ptr<scalar_t>());
+      auto range = at::arange(n, result.options());
```

Comment:
would using https://nvlabs.github.io/cub/classcub_1_1_counting_input_iterator.html#details instead of materializing arange be beneficial here?

Reply:
I don't think this works. Because according to https://nvlabs.github.io/cub/structcub_1_1_device_radix_sort.html, the argument is a pointer. I tried it and the compiler raises error 

```C++
/home/gaoxiang/pytorch-master/aten/src/ATen/native/cuda/TensorFactories.cu(106): error: no instance of overloaded function "cub::DeviceRadixSort::SortPairs" matches the argument list
            argument types are: (void *, size_t, int32_t *, int32_t *, cub::CountingInputIterator<int32_t, int>, scalar_t *, int64_t, int, unsigned long, c10::cuda::CUDAStream)
```

But I agree that this would be a good feature request for cub.